File: tatala-client-csharp/src/QiLeYuan/Tatala/executor/LocalTarget.cs
using System;
using System.Reflection;
using QiLeYuan.Tatala.socket.to;
using QiLeYuan.Tools.debug;

/**
 * This class is local target, simply provider local-method-call.
 * 
 * @author JimT
 *
 */
namespace QiLeYuan.Tatala.executor {

    public class LocalTarget : ServerTarget {
        public Object execute(TransferObject to) {

            String calleeClass = to.getCalleeClass();
            String calleeMethod = to.getCalleeMethod();

            Object retobj = null;
            try {
                Type type = Type.GetType(calleeClass);
                Object instance = Activator.CreateInstance(type);
                MethodInfo method = type.GetMethod(calleeMethod);
                retobj = method.Invoke(instance, new object[] { to });
            } catch (Exception e) {
                Logging.LogError(e.ToString());
            }

            return retobj;
        }
    }
}

File: history/0.1.2/tatala-java/src/com/qileyuan/tatala/executor/LocalTarget.java
package com.qileyuan.tatala.executor;

import java.lang.reflect.Method;

import com.qileyuan.tatala.socket.to.TransferObject;

/**
 * This class is local target, simply provider local-method-call.
 * 
 * @author JimT
 *
 */
public class LocalTarget implements ServerTarget{
	public Object execute(TransferObject to) {
		
		String calleeClass = to.getCalleeClass();
		String calleeMethod = to.getCalleeMethod();
		
		Object retobj = null;
		try {
			Class<?> cls = Class.forName(calleeClass);
			Object instance = cls.newInstance();
			Method meth = cls.getMethod(calleeMethod, TransferObject.class);
			retobj = meth.invoke(instance, to);
		} catch (Exception e) {
			e.printStackTrace();
		}
		
		return retobj;
	}
}
